--- $sourcebase.orig/tools/build/v2/user-config.jam
+++ $sourcebase/tools/build/v2/user-config.jam
@@ -45,6 +45,12 @@
 # Configure specific gcc version, giving alternative name to use.
 # using gcc : 3.2 : g++-3.2 ;
 
+# Configure gcc for cross compilation with MinGW32
+using gcc : 4.2.1 : $host_triplet-g++ : <rc>$host_triplet-windres <archiver>$host_triplet-ar ;
+toolset.add-requirements
+  <toolset>gcc:<target-os>windows
+  <toolset>gcc:<threadapi>win32 ;
+
 
 # -------------------
 # MSVC configuration.
